### PR TITLE
Fix a number of issues with the new room list's invites

### PR DESCRIPTION
**Pull request by turt2live**
Fixes https://github.com/vector-im/riot-web/issues/14309
Fixes https://github.com/vector-im/riot-web/issues/14358
For https://github.com/vector-im/riot-web/issues/13635

image
image

### CODE REVIEW

File: src/components/views/rooms/RoomSublist2.tsx [hunk 1]
```diff
@@ -311,15 +310,42 @@ export default class RoomSublist2 extends React.Component<IProps, IState> {
...
-
         let contextMenu = null;
         if (this.state.contextMenuPosition) {
             const isAlphabetical = RoomListStore.instance.getTagSorting(this.props.tagId) === SortAlgorithm.Alphabetic;
             const isUnreadFirst = RoomListStore.instance.getListOrder(this.props.tagId) === ListAlgorithm.Importance;
+
+            // Invites don't get some nonsense options, so only add them if we have to. We do
+            // this with an array instead of a containing div to ensure that the DOM structure
+            // is relatively sane.
+            let otherSections = [];
```
Comment (t3chguy) [inline src/components/views/rooms/RoomSublist2.tsx:321]: this feels like it should just be 

`otherSections = null;`

then `otherSections = <React.Fragment>...</React.Fragment>`
as there is no point dealing with keys

File: src/components/views/rooms/RoomSublist2.tsx [hunk 2]
```diff
@@ -311,15 +310,42 @@ export default class RoomSublist2 extends React.Component<IProps, IState> {
...
+            if (this.props.tagId !== DefaultTagID.Invite) {
+                otherSections.push(<hr key={otherSections.length} />);
```
Comment (t3chguy) [inline src/components/views/rooms/RoomSublist2.tsx:323]: key duplicate

File: src/components/views/rooms/RoomSublist2.tsx [hunk 3]
```diff
@@ -311,15 +310,42 @@ export default class RoomSublist2 extends React.Component<IProps, IState> {
...
+                otherSections.push(
+                    <div key={otherSections.length}>
```
Comment (t3chguy) [inline src/components/views/rooms/RoomSublist2.tsx:325]: key duplicate

File: src/components/views/rooms/RoomSublist2.tsx [hunk 4]
```diff
@@ -311,15 +310,42 @@ export default class RoomSublist2 extends React.Component<IProps, IState> {
...
+                        <div className='mx_RoomSublist2_contextMenu_title'>{_t("Unread rooms")}</div>
+                        <StyledCheckbox
+                            onChange={this.onUnreadFirstChanged}
+                            checked={isUnreadFirst}
+                        >
+                            {_t("Always show first")}
+                        </StyledCheckbox>
+                    </div>,
+                );
+                otherSections.push(<hr key={otherSections.length} />);
```
Comment (t3chguy) [inline src/components/views/rooms/RoomSublist2.tsx:335]: key duplicate

File: src/components/views/rooms/RoomSublist2.tsx [hunk 5]
```diff
@@ -311,15 +310,42 @@ export default class RoomSublist2 extends React.Component<IProps, IState> {
...
+                otherSections.push(
+                    <div key={otherSections.length}>
```
Comment (t3chguy) [inline src/components/views/rooms/RoomSublist2.tsx:337]: key duplicate

**Comment by turt2live** [comment]
> This also doesn't seem like it'll have the community invites following the correct sorting and they'll just be appended at the bottom in some random order?


At the top, intentionally. We're not putting effort into these.

File: src/components/views/rooms/RoomSublist2.tsx (hunk 2 above)
Comment (turt2live) [reply src/components/views/rooms/RoomSublist2.tsx:323]: These aren't duplicate, they go up by 1 naturally.

**Comment by t3chguy** [comment]
I think having a sorting function which doesn't work especially at the top end will be very confusing, especially for Matthew's usecase where he purposefully ignores some invites but wants to see the recent ones, if community ones block the recent ones that'll be confusing.

File: src/components/views/rooms/RoomSublist2.tsx (hunk 2 above)
Comment (t3chguy) [reply src/components/views/rooms/RoomSublist2.tsx:323]: oh, indeed, I'm blind

**Comment by turt2live** [comment]
Then we can stick them at the bottom or in their own section without ordering options. We are not putting effort into community invites.

**Pull request merged**